spec/support/shared_examples/mailer_shared_examples.rb

```diff
@@ -0,0 +1,28 @@
+shared_examples_for "multipart email" do
+  it "generates a multipart message (plain text and html)" do
+    expect(email.body.parts.length).to eq(2)
+    expect(email.body.parts.map(&:content_type)).to eq(["text/plain; charset=UTF-8", "text/html; charset=UTF-8"])
+  end
+end
+
+shared_examples_for "a translated email" do
+  it "does not have missing translations in HTML version" do
+    expect(email.html_part).not_to have_body_text("translation missing")
+  end
+
+  it "does not have missing translations in text version" do
+    expect(email.text_part).not_to have_body_text("translation missing")
+  end
+end
+
+shared_examples_for "an email with a valid sender" do
+  it "is delivered from Archive of Our Own <do-not-reply@example.org>" do
+    expect(email).to deliver_from("Archive of Our Own <#{ArchiveConfig.RETURN_ADDRESS}>")
+  end
+end
+
+shared_examples_for "a well formed HTML email" do
```

Comment:
Should we combine this with "multipart email"? If an email has HTML, it should be well-formed as well.

Reply:
I'm game. I left it separate in case we ever wanted to test that an email was multipart _and_ capable of handing a < or > sign in the text, but that seems somewhat unlikely... and if it ever becomes likely, well, we can just not use the "a multipart email" shared example in that particular test.